Snowflake.Data/Core/ArrowResultChunk.cs

```diff
@@ -123,106 +156,174 @@ public object ExtractCell(int columnIndex, SFDataType srcType, long scale)
                 case SFDataType.FIXED:
                     // Snowflake data types that are fixed-point numbers will fall into this category
                     // e.g. NUMBER, DECIMAL/NUMERIC, INT/INTEGER
-                    switch (column.Data.DataType.TypeId)
+                    switch (column)
                     {
-                        case ArrowTypeId.Int8:
+                        case Int8Array array:
+                            if (_sbyte[columnIndex] == null)
+                                _sbyte[columnIndex] = array.Values.ToArray();
                             if (scale == 0)
-                                return ((Int8Array)column).GetValue(_currentRecordIndex);
-                            else
-                                return ((Int8Array)column).GetValue(_currentRecordIndex) / (decimal)s_powersOf10[scale];
-                        case ArrowTypeId.Int16:
+                                return _sbyte[columnIndex][_currentRecordIndex];
+                            return _sbyte[columnIndex][_currentRecordIndex] / (decimal)s_powersOf10[scale];
```

Comment:
maybe some tests for the null values could be added

Reply:
Added more tests for null records.